src/clang-cpp-frontend/clang_cpp_adjust_expr.cpp

```diff
@@ -95,6 +95,7 @@ void clang_cpp_adjust::adjust_side_effect(side_effect_exprt &expr)
       destructor.arguments().push_back(address_of_exprt(new_object));
       expr.set("destructor", destructor);
     }
+    adjust_operands(expr);
```

Comment:
I think the pattern we use is to first check the operands. Does this work if you do it first? Otherwise, please add a comment explaining why the order is swapped

Reply:
Done